core/src/main/java/org/infinispan/configuration/cache/AsyncConfiguration.java

```diff
@@ -11,19 +14,32 @@
  */
 public class AsyncConfiguration {
 
-   private final boolean asyncMarshalling;
-   private final ReplicationQueue replicationQueue;
-   private final long replicationQueueInterval;
-   private final int replicationQueueMaxElements;
-   private final boolean useReplicationQueue;
-
-   AsyncConfiguration(boolean asyncMarshalling, ReplicationQueue replicationQueue, long replicationQueueInterval,
-         int replicationQueueMaxElements, boolean useReplicationQueue) {
-      this.asyncMarshalling = asyncMarshalling;
-      this.replicationQueue = replicationQueue;
-      this.replicationQueueInterval = replicationQueueInterval;
-      this.replicationQueueMaxElements = replicationQueueMaxElements;
-      this.useReplicationQueue = useReplicationQueue;
+   public static final AttributeDefinition<Boolean> MARSHALLING  = AttributeDefinition.builder("asyncMarshalling", false).immutable().build();
+   public static final AttributeDefinition<ReplicationQueue> REPLICATION_QUEUE  = AttributeDefinition.<ReplicationQueue>builder("replicationQueue", null, ReplicationQueue.class).immutable().build();
+   public static final AttributeDefinition<Long> REPLICATION_QUEUE_INTERVAL = AttributeDefinition.builder("replicationQueueInterval", 10l).build();
+   public static final AttributeDefinition<Integer> REPLICATION_QUEUE_MAX_ELEMENTS  = AttributeDefinition.builder("replicationQueueMaxElements", 1000).build();
+   public static final AttributeDefinition<Boolean> USE_REPLICATION_QUEUE = AttributeDefinition.builder("useReplicationQueue", false).immutable().build();
+
+   static final AttributeSet attributeDefinitionSet() {
+      return new AttributeSet(AsyncConfiguration.class, MARSHALLING, REPLICATION_QUEUE, REPLICATION_QUEUE_INTERVAL, REPLICATION_QUEUE_MAX_ELEMENTS, USE_REPLICATION_QUEUE);
+   }
+
+   private final Attribute<Boolean> marshalling;
+   private final Attribute<ReplicationQueue> replicationQueue;
+   private final Attribute<Long> replicationQueueInterval;
+   private final Attribute<Integer> replicationQueueMaxElements;
+   private final Attribute<Boolean> useReplicationQueue;
+
+   private AttributeSet attributes;
```

Comment:
Oh, sorry: we also need it when a *Builder reads() existing configuration. So we can't really do away with it.